src/yb/util/memory/arena.cc

```diff
@@ -253,13 +255,17 @@ void ArenaBase<Traits>::Reset(ResetMode mode) {
     // detection of memory-related bugs (invalid shallow copies, etc.).
     size_t last_size = current->full_size();
     current->Destroy(buffer_allocator_);
+    UntrackMemory(arena_footprint_);
     arena_footprint_ = 0;
     ReleaseStoreCurrent(nullptr);
     AddComponentUnlocked(NewBuffer(last_size, 0));
 #else
+    size_t bytes_before = arena_footprint_;
     arena_footprint_ = current->full_size();
+    TrackMemoryDifference(bytes_before, arena_footprint_);
```

Comment:
Replace by `TrackMemory`. Also remember to use `static_cast<int64_t>` when substracting the before and after value as the result might be negative.